R/source.R

```diff
@@ -46,13 +52,50 @@ source_file <- function(path, env = test_env(), chdir = TRUE,
       error = function(err) {
         abort(
           paste0("In path: ", encodeString(path, quote = '"')),
-          parent = err
+          parent = err,
+          call = error_call
         )
       }
     )
   }
 }
 
+filter_label <- function(exprs, label = NULL, error_call = caller_env()) {
+  if (is.null(label)) {
+    return(exprs)
+  }
+
+  found <- FALSE
+  include <- rep(FALSE, length(exprs))
+
+  for (i in seq_along(exprs)) {
+    expr <- exprs[[i]]
+
+    if (!is_call(expr, "test_that", n = 2)) {
+      include[[i]] <- TRUE
+    } else {
+      if (!is_string(expr[[2]]))
```

Comment:
Do you need to worry about people running code like `test_that({}, desc = "weird")`?

Reply:
If you do that you're on your own 😄 

People do weird things here (i.e. I bet someone does something like `test_that(my_fun("my label"), ...)`. If you want to handle those cases, IMO the IDE should figure out the line numbers from its parse tree and pass those through.